Sources/OpenAPIRuntime/Conversion/Converter+Server.swift

```diff
@@ -56,11 +56,34 @@ extension Converter {
                 // Drop everything after the optional semicolon (q, extensions, ...)
                 value.split(separator: ";")[0].trimmingCharacters(in: .whitespacesAndNewlines).lowercased()
             }
-
         if acceptValues.isEmpty { return }
-        if acceptValues.contains("*/*") { return }
-        if acceptValues.contains("\(substring.split(separator: "/")[0].lowercased())/*") { return }
-        if acceptValues.contains(where: { $0.localizedCaseInsensitiveContains(substring) }) { return }
+        guard let parsedSubstring = OpenAPIMIMEType(substring) else {
+            throw RuntimeError.invalidAcceptSubstring(substring)
+        }
+        guard case .concrete(let substringType, let substringSubtype) = parsedSubstring.kind else {
+            // If the substring content type has a wildcard, just let it through.
+            // It's not well defined how such a case should behave, so be permissive.
+            return
+        }
+
+        // Look for the first match.
+        for acceptValue in acceptValues {
+            // Fast path.
+            if acceptValue == substring { return }
+            guard let parsedAcceptValue = OpenAPIMIMEType(acceptValue) else {
+                throw RuntimeError.invalidExpectedContentType(acceptValue)
+            }
```

Comment:
How permissive would we like to be here? If a client sends an `Accept` header with a valid content-type that the server supports, and an invalid one, should the server permit it? IIUC, this `guard` statement suggests not, but is that the desired behaviour?

Reply:
If we fail to even parse the header value in its entirety, I suspect something very wrong is happening and I think we should fail, yeah. If we skip it silently, it could lead to harder-to-debug cases where the user is getting unexpected content back, and we're not catching that on the server (even though we could).

Or do you have cases in mind where this could be too strict?

Reply:
I think we need to consider the case where someone is calling the server from a not-generated client and might make mistakes and provide a malformed content-type followed by a valid one in the accept header. Having them get back a response of the valid one seems like a nicer experience than failing the request altogether, but I appreciate your opinion might differ here.

Reply:
Yeah I genuinely wonder, if they're typing in a malformed content type, whether they'd want us to fall back to the other way, or to error out (since there's probably a reason they added it in the first place).

Reply:
Discussed more offline, we'll leave it this strict, but separately I also filed https://github.com/apple/swift-openapi-generator/issues/644 so that we actually return 400, not 500 (as we'd return today when any error is thrown, including input validation error).